### PR TITLE
Liquidation refactor

### PR DESCRIPTION
**Pull request by kaichaosun**
The old liquidate pallet will remove later.
The liquidation using offchain worker still not sufficient for production, a client can be used by any community members should be much better.

### CODE REVIEW

File: pallets/liquidate-new/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+                }
+
+                let mut new_loans_detail = loans_detail.clone();
+                new_loans_detail.sort_by(|a, b| a.value.cmp(&b.value));
+                let liquidate_loans = &new_loans_detail[0];
+
+                if let Some(item) = collateral.1.iter().find(|collateral_item| {
+                    collateral_item.value
+                        >= T::LiquidateFactor::get().mul_floor(liquidate_loans.value)
+                }) {
```
Comment (MrPai) [inline pallets/liquidate-new/src/lib.rs:268]: Looks perfect. 
Here I'm thinking what if we have multi collateral currencies(such as DOT/KSM/BTC), but we only borrow USDT out, so the loan will only be USDT, in this situation, no single collateral's value will be bigger than the loans.
please fix me if I have any misunderstanding.

**Review by MrPai — COMMENTED** [review]
Great work.

File: pallets/liquidate-new/src/lib.rs (hunk 1 above)
Comment (MrPai) [reply pallets/liquidate-new/src/lib.rs:268]: #35

Comment (kaichaosun) [reply pallets/liquidate-new/src/lib.rs:268]: You are correct. Use offchain worker to deal this scenario is not flexible, we need a better solution to backup/replace this one.

**Comment by yz89** [comment]
Nice work!

**Pull request merged**